foundry.toml

```
@@ -0,0 +1,24 @@
+[profile.default]
+src = "src"
+out = "out"
+libs = ["node_modules", "lib"]
+remappings = [
+    "@ensdomains/=node_modules/@ensdomains/",
+    "@gnosis.pm/=node_modules/@gnosis.pm/",
+    "@openzeppelin/=node_modules/@openzeppelin/",
+    "@uniswap/=node_modules/@uniswap/",
+    "canonical-weth/=node_modules/canonical-weth/",
+    "eth-gas-reporter/=node_modules/eth-gas-reporter/",
+    "hardhat-deploy/=node_modules/hardhat-deploy/",
+    "hardhat/=node_modules/hardhat/",
+]
```

Are these remappings needed?
I'd leave them out until we actually start migrating how we import files.

```suggestion
remappings = []
```

---

I'm reticent to remove this as this was auto-generated by `forge init .` so foundry thought it necessary. In this configure, `forge build` works correctly.

---

Seems that remappings are needed _somewhere_ for the VS Code plugin to work for inspecting/making available introspection on dependency-related code.

---

This repo has very little dependencies, only Openzeppelin and only in tests.
The remapping autopopulation appears to match the result of the following find query, which is mostly from Node modules we don't use:

<details><summary>find -name '*.sol' -type f -exec grep --with-filename -E 'import "[^.]' {} \;</summary>

```
./src/contracts/test/TestERC20.sol:import "@openzeppelin/contracts/presets/ERC20PresetMinterPauser.sol";
./src/contracts/test/FeeClaimingERC20.sol:import "@openzeppelin/contracts/presets/ERC20PresetMinterPauser.sol";
./src/contracts/test/ERC20PresetPermit.sol:import "@openzeppelin/contracts/drafts/ERC20Permit.sol";
./src/contracts/test/ERC20PresetPermit.sol:import "@openzeppelin/contracts/presets/ERC20PresetMinterPauser.sol";
./src/contracts/test/vendor/ChiToken.sol:import "@openzeppelin/contracts/math/Math.sol";
./src/contracts/test/vendor/ChiToken.sol:import "@openzeppelin/contracts/math/SafeMath.sol";
./src/contracts/test/vendor/ChiToken.sol:import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
./node_modules/eth-gas-reporter/mock/test/TestMetacoin.sol:import "truffle/Assert.sol";
./node_modules/eth-gas-reporter/mock/test/TestMetacoin.sol:import "truffle/DeployedAddresses.sol";
./node_modules/@ensdomains/resolver/test/mocks/dummy.sol:import "@ensdomains/ens/contracts/ENSRegistry.sol";
./node_modules/@ensdomains/resolver/contracts/OwnedResolver.sol:import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
./node_modules/@ensdomains/resolver/contracts/PublicResolver.sol:import "@ensdomains/ens/contracts/ENS.sol";
./node_modules/@ensdomains/resolver/contracts/profiles/DNSResolver.sol:import "@ensdomains/dnssec-oracle/contracts/RRUtils.sol";
./node_modules/@ensdomains/resolver/contracts/DefaultReverseResolver.sol:import "@ensdomains/ens/contracts/ENS.sol";
./node_modules/@ensdomains/resolver/contracts/DefaultReverseResolver.sol:import "@ensdomains/ens/contracts/ReverseRegistrar.sol";
./node_modules/@gnosis.pm/safe-contracts/contracts/test/ERC20Token.sol:import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
./node_modules/@gnosis.pm/safe-contracts/contracts/test/Token.sol:import "@gnosis.pm/mock-contract/contracts/MockContract.sol";
./node_modules/hardhat/sample-projects/javascript/contracts/Lock.sol:// import "hardhat/console.sol";
./node_modules/hardhat/sample-projects/javascript-esm/contracts/Lock.sol:// import "hardhat/console.sol";
./node_modules/hardhat/sample-projects/typescript/contracts/Lock.sol:// import "hardhat/console.sol";
```
</details>


So I believe there should be no more than this:

```suggestion
remappings = [
    "@openzeppelin/=node_modules/@openzeppelin/",
]
```

But probably this is not needed as well if we set `"packageDefaultDependenciesDirectory": "node_modules"`. How can I see that our tool is not working?

---

Remappings have been completely eliminated.